### PR TITLE
[NEW] Provides ability to forward messages

### LINKED ISSUE #396: Ability forward messages into other chat rooms
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25351601-ability-forward-messages-into-other-chat-rooms?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by lindoelio**
@RocketChat/core 

Closes #396 

A simple and useful implement for provides the ability to forward messages into other chat rooms.

_Select the "Forward" option on message actions menu..._
screenshot from 2017-12-19 10-15-38

_Select at least one contact or channel that already subscribed and so keep forward..._
screenshot from 2017-12-19 10-29-35

### CODE REVIEW

File: packages/rocketchat-theme/client/imports/forms/switch.css [hunk 1]
```diff
@@ -49,9 +50,9 @@
 		transition: all 0.2s cubic-bezier(0.65, 0.05, 0.36, 1);
 
 		border-width: 1px;
-		border-color: var(--rc-color-error-light);
```
Comment (rodrigok) [inline packages/rocketchat-theme/client/imports/forms/switch.css]: Why did you change the current switch style?

File: packages/rocketchat-switch/rocketchat-switch.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<template name="switch">
```
Comment (rodrigok) [inline packages/rocketchat-switch/rocketchat-switch.html]: If you are creating a new component please replace the places where it is used to use the component

File: packages/rocketchat-ui-message/client/messagePreview.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<template name="messagePreview">
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/client/messagePreview.html]: Why do you need this component? Did you try to use the message render like we do here https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-integrations/client/views/integrationsIncoming.html#L119

**Comment by rodrigok** [comment]
I don't think that list of channels will be useful, why not use an input with autocomplete like we do in channel creation?

image

**Comment by rodrigok** [comment]
And, why send the message as user's message and not use quote or attachments to keep the information from origin? It could be optional

**Comment by cmrd-senya** [comment]
@lindoelio, are you planning to finish this PR? Or can I help you finishing it?

**Comment by cmrd-senya** [comment]
I started working on fixing this PR. I already rebased the work by @lindoelio to the latest develop in my fork: https://github.com/cmrd-senya/Rocket.Chat/tree/forward-messages

Now I'm going to work on implementing changes requested by @rodrigok

**Comment by ggazzo** [comment]
@cmrd-senya thanks, please if you need some help let me know.

**Comment by lawipac** [comment]
it seems message forwarding feature  is still not available in the latest rocket chat

**Comment by vins345** [comment]
Still not available.
